experimental/big_paper/plots.ipynb

```diff
@@ -1,1147 +1,1365 @@
+    "  # way.\n",
+    "  for m in split_metrics:\n",
+    "    df.loc[df['config.dataset'] == 'jft/entity:1.0.0', f'test_{m}'] = df.loc[\n",
+    "        df['config.dataset'] == 'jft/entity:1.0.0', f'val_{m}']\n",
+    "    df.loc[df['config.dataset'] == 'imagenet21k', f'test_{m}'] = df.loc[\n",
+    "        df['config.dataset'] == 'imagenet21k', f'val_{m}']\n",
+    "\n",
+    "  cols = ['model', 'config.dataset'] + metrics\n",
+    "  df = df[cols].copy()\n",
+    "  df = df.pivot(index='model', columns='config.dataset', values=metrics)\n",
+    "\n",
+    "  # Drop columns with all NaNs, e.g., ECE for JFT. They aren't measured.\n",
+    "  df = df.dropna(axis=1, how='all')\n",
+    "\n",
+    "  # Set few-shot imagenet metrics under a distinct dataset so later, we can\n",
+    "  # aggregate over few-shot metrics while excluding their original\n",
+    "  # config.dataset (the upstream dataset).\n",
+    "  for ds in fewshot_datasets:\n",
+    "    for f in [1, 5, 10, 25]:\n",
+    "      df[f'{f}shot_prec@1', f'few-shot {ds}'] = df[f'z/{ds}_{f}shot'].mean(axis=1)\n",
+    "      del df[f'z/{ds}_{f}shot']\n",
+    "  # Do same for compute and only keep upstream compute metrics.\n",
+    "  for metric in compute_metrics:\n",
+    "    for ds in df[metric]:\n",
+    "      if ds == 'imagenet21k':\n",
+    "        df[metric, 'compute'] = df[metric, ds]\n",
+    "      elif ds == 'jft/entity:1.0.0':\n",
+    "        df[metric, 'compute'] = np.where(df[metric, 'compute'].isnull(), df[metric, ds], df[metric, 'compute'])\n",
+    "      del df[metric, ds]\n",
+    "  return df\n",
+    "\n",
+    "def compute_score(df, datasets=None, categories=CATEGORIES):\n",
+    "  \"\"\"Compute aggregate score across metrics and per-category scores.\"\"\"\n",
+    "  df = df.copy()\n",
+    "\n",
+    "  # Scale all metrics in range [0.0, 1.0], and where higher is better.\n",
+    "  for column in df.columns:\n",
+    "    metric, dataset = column\n",
+    "    if 'ece' in metric:\n",
+    "      df[column] = 1. - df[column]\n",
+    "    if 'dataset' == 'compute':\n",
+    "      del df[column]\n",
+    "  # Remove 1-shot for now as its #s are unreliable due to high variance.\n",
+    "  del df['1shot_prec@1']\n",
+    "\n",
+    "  for metric, dataset in df[['test_loss', 'cifar_10h_loss', 'imagenet_real_loss']]:\n",
+    "    # Rescale NLL under its bound [0.0, uniform entropy]. Technically I21K &\n",
+    "    # JFT's uniform entropy should be computed on multiclass sigmoid NLL, but\n",
+    "    # unlike categorical uniform, multiclass sigmoid uniform is so large it's\n",
+    "    # meaningless as a bound.\n",
+    "    num_classes = DATASET_CLASSES[dataset]\n",
+    "    p = 1./num_classes\n",
+    "    max_value = -num_classes * p * np.log(p)\n",
+    "    df.loc[:, (metric, dataset)] = 1. - df[metric][dataset] / max_value\n",
+    "\n",
+    "  for metric, dataset in df[RETINA_NLL_METRICS]:\n",
```

Comment:
Is this code the same block as the one above? If so, It's better to define a local variable storing `['test_loss', 'cifar_10h_loss', 'imagenet_real_loss'] + RETINA_NLL_METRICS` and then to loop over the df subsetted to those columns.

same for compute_relative_score

Reply:
Not exactly the same because we have positive values of the NLL (and hence negative values of the LL) -- the normalization to the range[0, 1] requires computing a `min_value` in addition to the `max_value`.

Updated the code to reflect this.

Reply:
I see. It looks like you're grabbing both the min and max values based on the highest value in the dataframe. Is there not a theoretical min and max? In principle, the NLL should never be positive for discrete distributions.